common/bolt11.c

```diff
@@ -467,11 +467,11 @@ struct bolt11 *bolt11_decode(const tal_t *ctx, const char *str,
 
         b11->routes = tal_arr(b11, struct route_info *, 0);
 
-        if (!strstarts(str, "ln"))
-                return decode_fail(b11, fail, "Invoices must start with ln");
+        if (strlen(str) < 8)
+                return decode_fail(b11, fail, "Bad bech32 string");
 
-        hrp = tal_arr(tmpctx, char, strlen(str));
-        data = tal_arr(tmpctx, u5, strlen(str));
+        hrp = tal_arr(tmpctx, char, strlen(str) - 6);
```

Comment:
i realize this is a revert, so maybe this is already taken care of, but if not can you add some context about why these offsets are necessary, maybe as a code comment?

Reply:
It's actually documented in the bech32 reference code header to have these requirements.

The reason: bech32 includes a 30-bit (6char) checksum, and a `1` between hrp and data, meaning hrp must be 7 chars shorter.  Plus one for NUL terminator, so -6.

The `data` is not NUL-terminated, *and* the code insists that hrp be at least one letter long, so that's -8.

But you made me re-read the code and write some test cases to be *sure*!